testing/cpm/cpm_generate_pins-nested/CMakeLists.txt

```diff
@@ -24,8 +24,15 @@ add_subdirectory(b)
 include(${rapids-cmake-dir}/cpm/init.cmake)
 rapids_cpm_init(GENERATE_PINNED_VERSIONS)
 
+# only check projects that were downloaded by CPM (ignore those already in the build environment)
+#
 # don't check cuco here as it uses hashes and that breaks our verify script
-set(projects-to-verify rmm fmt spdlog CCCL)
+foreach(proj IN ITEMS rmm fmt spdlog CCCL)
+  if(${proj}_SOURCE_DIR)
+    list(APPEND projects-to-verify ${proj})
+  endif()
+endforeach()
```

Comment:
I chose not to add validation here that `projects-to-verify` is non-empty. That already happens in the `verify/` scripts.

e.g.

https://github.com/rapidsai/rapids-cmake/blob/e1fbcd956fa9184c23860747410f0bda6251b2f5/testing/cpm/cpm_generate_pins-nested/verify/CMakeLists.txt#L25-L28